use-cases/core/median-finding/median_finding.cpp

```diff
@@ -0,0 +1,186 @@
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions Contact  
+//
+// ************************************************************************
+//@HEADER
+*/
+
+// The simple model creates n points in 1d evenly distributed 0..1,
+// then coordinates are squared so half the points lie below 0.25.
+// A thread reduce loop inside a team reduce loop checks each coordinate
+// and determines if they are left or right of the current cut.
+// The total weights to the left of the cut are reduced (summed).
+// Then the cut is shifted to estimate a new cut position,
+// where the goal is to have the left and right have the same weight.
+// The main loop runs the full test for a range of team counts
+// and logs the time cost for each.
+
+#include <Kokkos_Core.hpp>
+
+// store results from a run
+struct Result {
+  int time_ms;        // time in ms for actual partition
+  double cut_line;    // the final cut
+  int n_teams;        // how many teams we ran with
+};
+
+// in this simple model N points is restricted to
+// be a power of 2 and we test N teams in powers
+// of 2 as well so it's always an even factor.
+#define KOKKOS_N_POINTS_POW_OF_2 23
+
+// main will call this simple_model test for varies n_teams
+Result simple_model(int n_teams) {
+
+    // Fix the total number of points
+    const int n_points = pow(2,KOKKOS_N_POINTS_POW_OF_2); // keep this in int bounds or change type
+    const int n_points_per_team = n_points / n_teams;
+
+    // currently setup assuming n_teams will be a factor
+    if(n_points_per_team * n_teams != n_points) {
+      throw std::logic_error("For now this assumes n_points is a multiple of n_teams.");
+    }
+
+    // for now simple epsilon assumes no duplicate points (should probably have a tiny epsilon for rounding error)
+    double epsilon = 0.5; // the final weight should not deviate from the perfect weight by more than 1/2 a weight unit
+
+    // make the views using the default device
+    typedef Kokkos::View<double*> view_t;
+    view_t x("x", n_points);
+
+    // will fix team count and use Kokkos::AUTO() - this policy is the only one used in the model
+    auto policy = Kokkos::TeamPolicy<>(n_teams, Kokkos::AUTO);
+    typedef typename Kokkos::TeamPolicy<>::member_type member_type;
+
+    // initialize the coordinates in a parallel_for - we want UVM off to work
+    Kokkos::parallel_for(n_points, KOKKOS_LAMBDA (int i) {
```

Comment:
Label your kernels for better profiling output :-)

`Kokkos::parallel_for ("my kernel name", n_points, ...)`

Reply:
Updated. Thanks for feedback.